api/src/services/watchListService.ts

```diff
@@ -0,0 +1,156 @@
+import { WatchListParser } from "../models/watchList";
+import TMDBRepository from "../repositories/tmdbRepository";
+
+export default class WatchListService {
+    private watchListRepository: Repository<WatchListEntity>;
+    private watchListItemRepository: Repository<WatchListItemEntity>;
+    private userRepository: Repository<UserEntity>;
+    private mediaRepository: Repository<MediaEntity>;
+    private tmdbRepository: TMDBRepository;
+
+    public constructor(
+        watchListRepository: Repository<WatchListEntity>,
+        watchListItemRepository: Repository<WatchListItemEntity>,
+        userRepository: Repository<UserEntity>,
+        mediaRepository: Repository<MediaEntity>,
+        tmdbRepository: TMDBRepository
+    ) {
+        this.watchListRepository = watchListRepository;
+        this.watchListItemRepository = watchListItemRepository;
+        this.userRepository = userRepository;
+        this.mediaRepository = mediaRepository;
+        this.tmdbRepository = tmdbRepository;
+    }
+
+    public async createWatchList(
+        userId: number,
+        title: string,
+        description: string,
+        privacyType: WatchListPrivacyType
+    ) {
+        if (ValidationUtils.isEmpty(title))
+            throw new EmptyWatchListTitleError();
+        if (ValidationUtils.isEmpty(description))
+            throw new EmptyWatchListDescriptionError();
+
+        const userExists = await this.userRepository.exist({
+            where: { Id: userId },
+        });
+
+        if (!userExists) throw new UserNotExistsError();
+
+        const watchList = await this.watchListRepository.save({
+            Owner: {
+                Id: userId,
+            },
+            Title: title,
+            Description: description,
+            PrivacyType: privacyType,
+        } as unknown as WatchListEntity);
+
+        return watchList;
+    }
+
+    public async addWatchListItems(
+        userId: number,
+        watchListId: number,
+        mediaIds: number[]
+    ) {
+        await this.validateAddWatchListItemsArguments(userId, watchListId);
+        const promises = mediaIds.map((mediaId) =>
+            this.saveMediaIntoWatchList(watchListId, mediaId)
+        );
+        return await Promise.all(promises);
+    }
+
+    public async getWatchListItems(
+        userId: number | undefined,
+        watchListId: number
+    ) {
+        if (userId) {
+            const userExists = await this.userRepository.exist({
+                where: { Id: userId },
+            });
+            if (!userExists) throw new UserNotExistsError();
+        }
+
+        const watchList = await this.watchListRepository.findOne({
+            where: { Id: watchListId },
+            relations: ["Owner", "WatchListItems", "WatchListItems.Media"],
+        });
+        if (!watchList) throw new WatchListNotFoundError();
+
+        const isOwner = watchList.Owner.Id === userId;
+        const isFriend = true; // Insert friendship logic here
```

Comment:
Aqui, acho que fica melhor criar um método `areFriends(owner, userId)` e deixar ele retornando `true` por enquanto. Aí fica fácil modificar depois quando o #45 for aprovado.